californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java

```diff
@@ -586,6 +595,10 @@ public void removeInterceptor(final MessageInterceptor interceptor) {
 
 	@Override
 	public void sendRequest(final Request request) {
+		if (!running) {
+			request.cancel();
+			return;
+		}
```

Comment:
With a different PR addressing the `stop` behaviour (callbacks to the requests), I'm much more afraid that the "continuous merging" introduces error. Therefore I added this here. Without, and using a "detached executor" in the server or client, it may also cause unexpected logging messages/exceptions.